tests/test_campaign.py

```diff
@@ -73,3 +80,34 @@ def test_candidate_toggling(constraints, exclude, complement):
     other = campaign._searchspace_metadata[_EXCLUDED].drop(index=idx)
     assert all(target == exclude)  # must contain the updated values
     assert all(other != exclude)  # must contain the original values
+
+
+@pytest.mark.parametrize(
+    "flag",
+    [
+        "allow_recommending_already_measured",
+        "allow_recommending_already_recommended",
+        "allow_recommending_pending_experiments",
+    ],
+    ids=lambda x: x.removeprefix("allow_recommending_"),
+)
+@pytest.mark.parametrize(
+    "space_type",
+    [SearchSpaceType.DISCRETE, SearchSpaceType.CONTINUOUS],
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("value", [True, param(UNSPECIFIED, id=repr(UNSPECIFIED))])
```

Comment:
The test was meant as "boolean vs unspecified" hence only one of the booleans was used. And because our test suite is already way too unorganized, I did't add the second boolean. But since we are talking only about a hundredth of a second of overhead, let me add it as well for the sake of completeness 👍🏼